app/models/concerns/advanced_search_condition_dispatcher.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+# Shared operator dispatch for advanced search conditions.
+#
+# This concern expects the including class to:
+# - include `AdvancedSearchConditions` (for `build_arel_node` and `condition_*` helpers)
+# - provide `model_class`
+#
+# Override points:
+# - `normalize_condition_value(condition)` to coerce values (e.g., enums)
+# - `apply_*_operator` methods for special-case fields (e.g., upcasing PUID)
+module AdvancedSearchConditionDispatcher
```

Comment:
As discussed please change the name of this concern so that it makes it clear of what it is